projects/ngx-quill/src/lib/quill-editor.component.ts

```diff
@@ -220,7 +221,16 @@ export abstract class QuillEditorBase implements AfterViewInit, ControlValueAcce
     // this will lead to runtime exceptions, since the code will be executed on DOM nodes that don't exist within the tree.
 
     this.quillSubscription = this.service.getQuill().pipe(
-      mergeMap(Quill => this.service.registerCustomModules(Quill, this.customModules))
+      mergeMap((Quill) => {
+        const promises = [this.service.registerCustomModules(Quill, this.customModules)]
+        if (this.beforeRender) {
+          promises.push(this.beforeRender())
+        }
+        if (this.service.config.beforeRender) {
```

Comment:
i would make the "beforeRender" at the component overriding the service.beforeRender.

All inputs and configs are currently working this way. What do you think?

Reply:
and tests would be nice :)